lib/compress/zstd_compress.c

```diff
@@ -1542,7 +1577,7 @@ size_t ZSTD_estimateCCtxSize_usingCCtxParams(const ZSTD_CCtx_params* params)
      * be needed. However, we still allocate two 0-sized buffers, which can
      * take space under ASAN. */
     return ZSTD_estimateCCtxSize_usingCCtxParams_internal(
```

Comment:
Note : 
I find it strange that `ZSTD_estimateCCtxSize_usingCCtxParams_internal()` doesn't actually use a `const ZSTD_CCtx_params*` object as parameter.
Instead, it uses a long and growing list of parameters, some of them extracted selectively from `const ZSTD_CCtx_params*`.

It seems that accepting a `const ZSTD_CCtx_params*` main parameter would cut down some of the complexity.

That being said, I understand that refactoring `ZSTD_estimateCCtxSize_usingCCtxParams_internal()`, an existing function, is not necessarily part of this PR.
But maybe this could be a legitimate follow up.

Reply:
If it's trivial I'll do it in this PR, otherwise I'll file an issue to track this work.

Reply:
Filed an issue. I think it's a great one for new contributors to take on! https://github.com/facebook/zstd/issues/3404